### PR TITLE
add end-level models, remove end-game logic calculation properties from passageDS

### LINKED ISSUE #18: add individual end-level result passage model

### LINKED ISSUE #19: add group end-level result passage model

### PR DESCRIPTION
**Pull request by tongxiang**
#### What's this PR do?
1. Adds `passageEndLevelIndiv.js` and `passageEndLevelGroup.js` models, as well as all the corresponding supporting code 
2. Removes `indivSuccessPath` and `indivSuperlativePath` properties from `passageDS` and adds them to `passageEndLevelIndiv`
3. Removes `groupSuccessPath` property from `passageDS` and adds it to `passageEndLevelGroup`
4. Rearranges the order in which the buttons for adding passages appears on the add passage toolbar
5. Removes Mercurial files
#### Where should the reviewer start?

In adding new models, I followed the following rough order of tasks to be done. It might provide a helpful way of seeing all the changes/additions allowing the new models to function. 
1. new passage editing objects, extended off of storyeditview
2. new html files which are the new modals to be edited
3. add new editor to storyeditview.js 
4. add a new control flow statement to js/collections/passagecollection.js 
5. add the new script tags to app.html
6. add the new passage node type to templates/storyeditview/toolbar.html
7. add corresponding toolbar javascripts to toolbar.js
8. add reference in edit function in passageitemview.js
9. change storyeditview.html, files baked
#### How should this be manually tested?

Created a game, inputted data to every model type, ensured that data inputted to every model type was retained if story was re-opened, published a file and confirmed that the HTML source code retained all the inputted data. 
#### What are the relevant tickets?

Closes #19, closes #18
#### Questions:

Added issues for this, but would it be possible for subclasses to inherit copy-and-pasted properties `passageDS.js` and `passageDSEditor.js` models?

### CODE REVIEW

**Comment by tongxiang** [comment]
#### What's this PR do?
1. Adds `passageEndLevelIndiv.js` and `passageEndLevelGroup.js` models, as well as all the corresponding supporting code 
2. Removes `indivSuccessPath` and `indivSuperlativePath` properties from `passageDS` and adds them to `passageEndLevelIndiv`
3. Removes `groupSuccessPath` property from `passageDS` and adds it to `passageEndLevelGroup`
4. Rearranges the order in which the buttons for adding passages appears on the add passage toolbar
5. Removes Mercurial files

#### Where should the reviewer start?
In adding new models, I followed the following rough order of tasks to be done. It might provide a helpful way of seeing all the changes/additions allowing the new models to function. 

   1. NEW PASSAGE EDITING OBJECTS, EXTENDED OFF OF STORYEDITVIEW
   2. NEW HTML FILES WHICH ARE THE NEW MODALS TO BE EDITED
   3. ADD NEW EDITOR TO STORYEDITVIEW.JS 
   4. ADD A NEW CONTROL FLOW STATEMENT TO JS/COLLECTIONS/PASSAGECOLLECTION.JS 
   5. ADD THE NEW <script> TO APP.HTML
   6. ADD THE NEW PASSAGE NODE TYPE TO TEMPLATES/STORYEDITVIEW/TOOLBAR.HTML
   7. ADD CORRESPONDING TOOLBAR JAVASCRIPTS TO TOOLBAR.JS
   8. ADD REFERENCE IN EDIT FUNCTION IN PASSAGEITEMVIEW.JS
   9. CHANGE STORYEDITVIEW.HTML, FILES BAKED

#### How should this be manually tested?
Created a game, inputted data to every model type, ensured that data inputted to every model type was retained if story was re-opened, published a file and confirmed that the HTML source code retained all the inputted data. 

#### What are the relevant tickets?
Closes #19, closes #18

#### Questions:
Added issues for this, but would it be possible for subclasses to inherit copy-and-pasted properties `passageDS.js` and `passageDSEditor.js` models?

File: js/views/storyeditview/passageEndLevelIndivEditor.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    this.$el.data('modal').trigger('show');
+  },
+
+
+  /**
+   * Closes modal.
+   */
+  close: function()
+  {
```
Comment (jonuy) [inline js/views/storyeditview/passageEndLevelIndivEditor.js]: Move `{` to the same line as `close: function()`

File: js/views/storyeditview/passageEndLevelGroupEditor.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+    this.$el.data('modal').trigger('show');
+  },
+
+  /**
+   * Closes modal.
+   */
+  close: function()
+  {
```
Comment (jonuy) [inline js/views/storyeditview/passageEndLevelGroupEditor.js]: Move `{` to the same line as `close: function()`

File: js/views/storyeditview/passageEndLevelGroupEditor.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   */
+  open: function() {
+    this.$('.passageId').val(this.model.id);
+    this.$('.passageName').val(this.model.get('name'));
+    var text = this.model.get('text');
+    this.$('.passageText').val((text == Passage.prototype.defaults.text) ? '' : text);
+    this.$('#edit-ds-oip').val(this.model.get('optinpath'));
+    this.$('#edit-group-success-path').val(this.model.get('groupSuccessPath'));
+
+
```
Comment (jonuy) [inline js/views/storyeditview/passageEndLevelGroupEditor.js]: There's an extra new line here

**Comment by jonuy** [comment]
Only nitpicking on formatting. :+1:

**Pull request merged**